blockchain/block/runnable.go

```diff
@@ -43,6 +45,11 @@ func (b *RunnableActionsBuilder) AddActions(acts ...action.SealedEnvelope) *Runn
 
 // Build signs and then builds a block.
 func (b *RunnableActionsBuilder) Build() RunnableActions {
-	b.ra.txHash = calculateTxRoot(b.ra.actions)
+	var err error
+	b.ra.txHash, err = calculateTxRoot(b.ra.actions)
+	if err != nil {
+		log.L().Debug("error in getting hash ", zap.Error(err))
```

Comment:
concerning

Reply:
@dustinxie @CoderZhi  Can you please confirm if tests are failing because of product defect or any other change is required in this PR ?  Thanks.

Reply:
test is failing, need to fix: `TestBlockListener: blocklistener_test.go:49`: (click on failed CI test `Details`, you can see that)

other than that, it looks good. Previously, Hash() would panic if sth goes wrong so no error is returned, now an error is returned as a graceful way of failing.

Reply:
@ashishsnigam  also would you like to pick up https://github.com/iotexproject/iotex-core/issues/2749?

Reply:
Yes for #2749 and need help for test fixing. If you can give me some pointers or reference of any other test. @dustinxie

Reply:
i think it's b/c there's no actions in the block so it hits `if len(h) == 0`
you can add a dummy action before calling `builder.SignAndBuild` to fix the error

Reply:
also you can run `make test` to run the CI test locally, and see particular error message

Reply:
also pls run `make fmt` before submit PR, thx

Reply:
sure, can u give me some pointer on how to add dummy action ?

Reply:
I added a dummy action blocklistner_test.go#46, #54 and tested it locally. Testcase passed. If this is not what you suggested, please give me pointer.

Reply:
Local test run revealed that same thing might need to be done in multiple places. Now I need two inputs - 
1. how to create action with zero hash (quick fix)
2. Domain knowledge pointer when to use which action. @dustinxie

Reply:
see my new comment